hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/repository/CustomFeeRepository.java

```diff
@@ -17,6 +17,23 @@
 package com.hedera.mirror.web3.repository;
 
 import com.hedera.mirror.common.domain.token.CustomFee;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
 
-public interface CustomFeeRepository extends CrudRepository<CustomFee, Long> {}
+public interface CustomFeeRepository extends CrudRepository<CustomFee, Long> {
+    @Query(
+            value =
+                    """
+            select * from (
+                select * from custom_fee where token_id = ?1
+                union all
+                select * from custom_fee_history where token_id = ?1
+            ) as cf
+            where timestamp_range < int8_range(?2, null)
```

Comment:
Can use lower and <=. Also need to push this into the sub-queries instead of having in the outer.

```suggestion
            where lower(timestamp_range) <= ?2
```